DI 2086 tso500 s2 verification

Arun-bioinformatics opened the pull request:
Python scripts implemented in TSO500 S2 verification, confuence page can be found here:
https://cuhbioinformatics.atlassian.net/wiki/spaces/URA/pages/3911057569/DRAFT+2025079+TSO500+S2+verifiation

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)
<!-- Reviewable:end -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - CLI to enrich a features CSV with depth (DP), allele frequency (VAF) and filter status from two VCFs, producing a consolidated CSV.
  - Interactive correlation plot comparing VAF_truth and VAF_query, colour‑coded by filter‑change categories, with hover details, y=x reference line and export to HTML/PDF.
  - Automated per‑run stacked bar charts showing shared, sample‑only and run‑only variant counts, plus a saved summary CSV.

- Bug Fixes
  - Improved input validation, diagnostic messages and user‑facing logs for easier troubleshooting.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
Correlation plot script pdf added

Commit:
Upload exctract_vaf_dp_filter.py file to create .merged.csv files

Commit:
Upload the sompy_plots.py
- This script can be run on its own.

Commit:
Improve the colour consitency of variants in correlation plots

RSWilson1 commented on DI-2086/extract_vaf_dp_filter.py:
```diff
@@ -0,0 +1,289 @@
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] bcftools query failed: {e.stderr}")
+    return 0, 0.0, '.'
+
```
Could you try this and see if it changes the results?
<!-- Reviewable comment -OZikIOl4YTZNObZRbHS:brqxth -->
<!-- Sent from Reviewable.io -->

RSWilson1 replied:
I have had a think about this one and it is wrong, the quotes are used for the bcftools cmd. Many examples shown here: https://samtools.github.io/bcftools/bcftools.html#query.
<!-- Reviewable comment -O_RGZ735dEkGnMyI2QI:bqbmba8 -->
<!-- Sent from Reviewable.io -->

RSWilson1 replied:
Pandas docs show that this code does work as column names provided
<!-- Reviewable comment -O_ul8bi2gezX8nABbqd:b34abbd -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
@coderabbitai[bot], even if the conventional order is CHROM, POS, REF, ALT, FILTER, DP, VF. There is a bcftools query above which uses the alternate order which aligns with the columns named in DataFrame (i.e. CHROM POS REF ALT DP VF FILTER)
<!-- Reviewable comment -O_w6i1L-WnzCBtIF0oM:b3yhe96 -->
<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> @RSWilson1 reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 24 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/extract_vaf_dp_filter.py` line 57 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZidDYvBCXNna0gN98I:-OZidDYvBCXNna0gN98J:b-gzn3t4) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L57)):*
> ```Python
>             # Skip rows where we still don't have ref/alt after trying fallbacks
>             if not ref or not alt:
>                 print(f"Warning: Skipping row at {chrom}:{pos} - missing REF/ALT data")
> ```

Not sure we want this, is this something that came up?
___
*[`DI-2086/extract_vaf_dp_filter.py` line 105 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZih8AH-Wko5dR_2SyZ:-OZih8AIBst9eo3xzVwO:b-jvcx70) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L105)):*
> ```Python
>     base_name = "-".join(sample_name.split("-")[:2])
>     print(f"Finding VCF files for base name: {base_name}")
>     s2_vcf = glob.glob(f"query_filtered_VCFs/S2_VCF/{base_name}*s2*.vcf.gz")
> ```

would be nice to not have this hard-coded and generic for processing different validations
___
*[`DI-2086/extract_vaf_dp_filter.py` line 109 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihOQeEV4DAJd9HFqK:-OZihOQeEV4DAJd9HFqL:bbf968i) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L109)):*
> ```Python
> 
>     # Remove s2 from s1 list
>     s1_vcf = [f for f in s1_vcf if "s2" not in f]
> ```

not needed? and specific to S2 validation
___
*[`DI-2086/extract_vaf_dp_filter.py` line 284 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihaF0CtNZWSC1IsuK:-OZihaF0CtNZWSC1IsuL:bcdkah) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L284)):*
> ```Python
> if __name__ == "__main__":
>     parser = argparse.ArgumentParser(
>         description="Extract VAF and DP from VCF files based on features CSV."
> ```

Could these be moved into parse_args function and main updated for using that.
___
*[`DI-2086/sompy_plots.py` line 14 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihrjsCTAYnWJPOOPt:-OZihrjsCTAYnWJPOOPu:bnqhl6i) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L14)):*
> ```Python
> from io import StringIO
> 
> dxfile = dxpy.DXFile('file-J24FXZQ48fFz1gb8J1Jxg1Vq').read()
> ```

hard-coded
___
*[`DI-2086/sompy_plots.py` line 19 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihu5S-zrYAKymzwqm:-OZihu5S-zrYAKymzwqn:bnqhl6i) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L19)):*
> ```Python
> 
> # Select the project
> project = "project-J1zbgv84kZ75y8B1vzXv6X2v"
> ```

hard-coded
___
*[`DI-2086/sompy_plots.py` line 27 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihuru0_BM-PXyw_Bg:-OZihuru0_BM-PXyw_Bh:bnqhl6i) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L27)):*
> ```Python
>     name_mode="glob",
>     project=project,
>     folder="/query_filtered_VCFs_removed_sompy_results"
> ```

hard-coded
___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-OZibXLf994j-etY7zBi:blsfc5t) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
> ```Python
> 
>     Returns:
>         pd.DataFrame: The modified dataframe with the new 'FILTER_change' column.
> ```

still called dataframe so should be in docstring here
___
*[`DI-2086/correlation_plots_pdf_html.py` line 49 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZifqFZB9Mqy2MJQyqJ:-OZifqFZB9Mqy2MJQyqK:b-8dmsgb) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L49)):*
> ```Python
>     ]
>     
>     dataframe['FILTER_change'] = np.select(conditions, choices, default='Unknown')
> ```

Do we get many unknowns and could we handles these better?
___
*[`DI-2086/correlation_plots_pdf_html.py` line 57 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibdTZB5iT_dbzb1Q1:-OZibdTZB5iT_dbzb1Q2:b3kre) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L57)):*
> ```Python
>     Create an interactive correlation plot with VAF_s1 vs VAF_s2 colored by tag
>     with CHROM information displayed on hover
>     #TO BE CHANGED
> ```

?
___
*[`DI-2086/correlation_plots_pdf_html.py` line 61 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibg-WAqIND5BEqxw7:-OZibg-WAqIND5BEqxw8:b-hlba3h) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L61)):*
> ```Python
>     Todo's
>     - Print out the unique values for the filter column to verify possible combination
>     - Write a function that adds a new column based on the conditions specified
> ```

Were these points done? should it be here?
___
*[`DI-2086/correlation_plots_pdf_html.py` line 69 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZigHJ98W-FS2avoWay:-OZig9_r41-0iqsKOXow:bkybedt) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L69)):*
> ```Python
>         df = pd.read_csv(csv_file)
>         
>         df = add_filter_column_change(df)
> ```

could we change these variables to have better names and change based on what has been done?



_Code quote:_
```Python
        df = pd.read_csv(csv_file)

        df = add_filter_column_change(df)
```
___
*[`DI-2086/correlation_plots_pdf_html.py` line 203 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZicA148GxYhOiIxb8V:-OZicA148GxYhOiIxb8W:bers8l7) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L203)):*
> ```Python
> 
> if __name__ == "__main__":
>     parser = argparse.ArgumentParser(
> ```

could these be moved into a function and then call that function in main and just have main() here. Main would no longer require the inputs as well.


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 27 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-OZn6fPE6ENpkGs1F9uI:b-896fix) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

still called dataframe so should be in docstring here
</blockquote></details>

Done.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 151 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnIaLS7dt2ot1vxIaw:-OZnIaLS7dt2ot1vxIax:b-vi9tyn) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L151)):*
> ```Python
>     Uses bcftools query to extract variant information. If the variant is not
>     found or bcftools fails, returns (0, 0.0) as default values.
>     """
> ```

Remove duplicates in the feature.csv file? As multiple entries presenting the same variant was found in the .vcf file of origin and .features.csv file sompy output, this is because there are multiple annotations for more than one transcript in the sompy input vcf.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 165 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnJZEn946UiOEUIeOy:-OZnJZEn946UiOEUIeOz:b-jkovn9) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L165)):*
> ```Python
>                 f"{chrom}:{pos}-{pos}",
>                 "-f",
>                 "'%CHROM\t%POS\t%REF\t%ALT\t%FILTER\t%DP\t[%VF]\n'",
> ```

Switch to Format DP (i.e [%DP]) instead.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 185 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnL6gLApGtkF1wP9ct:-OZnL6gMCugDndQ4Ek2t:bnm7fei) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L185)):*
> ```Python
>     except subprocess.CalledProcessError as e:
>         print(f"[ERROR] bcftools query failed: {e.stderr}")
>     return 0, 0.0, '.'
> ```

Change the method in the future to join features dataframe with the vcf relevant entries. Error handling or edge cases need to be addressed in another way as well.


<!-- Sent from Reviewable.io -->

Commit:
In extract_vaf_dp_filter.py
- Improve Docstrings for every function
- New inputs required:
    - query_vcf_file
    - truth_vcf_file
    - sample_name
- Move the parse_args() function out of the if __name__ == 'main' statement
- Changed name function parse_features_file to read_features_file
- Changed read_features_file to output a pd.DataFrame
- find_vcf_files function deleted, including the hardcoded paths
- Changed the function of extract_variant_data_df
    - Improved speed and performance
    - bcftools query now returns all variants instead of a single variants
- new function responsible of merging records to vcfs' data (merge_records_to_query_truth())
- write_output function removed
- Replaced s1 and s2 with truth and query respectively
- Addressed PEP8 issues

In correlation_plots_pdf_html.py
- Improve Docstrings for every function
- Added new conditions to mark variants which have been "rescued"
- Colour correction on the correlation plots
- Replaced s1 and s2 with truth and query respectively
- Addressed PEP8 issues

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 26 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 49 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZifqFZB9Mqy2MJQyqJ:-O_NDyBJ1MwZYwSLn9LP:b-23yuv1) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L49)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Do we get many unknowns and could we handles these better?
</blockquote></details>

I have added new conditions to mark the variants that are rescued, as those were the ones that were marked as 'unknown'. However I will still keep this line in case there are some scenarios that I have not taken into account if the code is reused again.
___
*[`DI-2086/correlation_plots_pdf_html.py` line 57 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibdTZB5iT_dbzb1Q1:-O_NEDHR-1nc9B5OWlsp:b-gqbjka) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L57)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

?
</blockquote></details>

Docstring updated for this
___
*[`DI-2086/correlation_plots_pdf_html.py` line 61 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibg-WAqIND5BEqxw7:-O_NEGX01rF7OaAgHH6V:b-al74r6) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L61)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Were these points done? should it be here?
</blockquote></details>

They were done
___
*[`DI-2086/correlation_plots_pdf_html.py` line 203 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZicA148GxYhOiIxb8V:-O_NESeh3jprGEqUmc_h:b-fu5zj8) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L203)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

could these be moved into a function and then call that function in main and just have main() here. Main would no longer require the inputs as well.
</blockquote></details>

Moved into funciton
___
*[`DI-2086/extract_vaf_dp_filter.py` line 57 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZidDYvBCXNna0gN98I:-O_NEVm_72Nf4-6lg4a-:b2napec) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L57)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Not sure we want this, is this something that came up?
</blockquote></details>

Method has changed, so that line has been deleted.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 105 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZih8AH-Wko5dR_2SyZ:-O_NEanx0djSmxRKeQsK:bo98fbw) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L105)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

would be nice to not have this hard-coded and generic for processing different validations
</blockquote></details>

Hard-coded step removed.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 109 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihOQeEV4DAJd9HFqK:-O_NEdooEfsNp8n0pJqj:bhofej8) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L109)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

not needed? and specific to S2 validation
</blockquote></details>

Deleted
___
*[`DI-2086/extract_vaf_dp_filter.py` line 151 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnIaLS7dt2ot1vxIaw:-O_NEl3e1E_FgXXEQSNZ:bb6un1j) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L151)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Remove duplicates in the feature.csv file? As multiple entries presenting the same variant was found in the .vcf file of origin and .features.csv file sompy output, this is because there are multiple annotations for more than one transcript in the sompy input vcf.
</blockquote></details>

Done
___
*[`DI-2086/extract_vaf_dp_filter.py` line 185 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnL6gLApGtkF1wP9ct:-O_NEnko4wYXMb1p4Kwc:b-h1jj8m) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L185)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Change the method in the future to join features dataframe with the vcf relevant entries. Error handling or edge cases need to be addressed in another way as well.
</blockquote></details>

Method changed
___
*[`DI-2086/extract_vaf_dp_filter.py` line 284 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihaF0CtNZWSC1IsuK:-O_NF2__CMfW44kohABQ:bb7f2mf) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L284)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Could these be moved into parse_args function and main updated for using that.
</blockquote></details>

done


<!-- Sent from Reviewable.io -->

Commit:
Use FORMAT/DP instead of INFO/DP
- Updated %DP to [%DP]

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 27 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 69 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZigHJ98W-FS2avoWay:-O_NNAn08EPrQ0B5ceV0:b-1fwpsl) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L69)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

could we change these variables to have better names and change based on what has been done?
</blockquote></details>

I made some minor changes to the variables, but because the df variant is used a lot within the same function, I decided to keep it as it is.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 165 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnJZEn946UiOEUIeOy:-O_NNPKi9NCGx0mDS6cc:bb6un1j) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L165)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Switch to Format DP (i.e [%DP]) instead.
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

RSWilson1 commented on DI-2086/correlation_plots_pdf_html.py:
```diff
@@ -0,0 +1,210 @@
+            print(f"Error: Missing columns: {missing_cols}")
+            print(f"Available columns: {list(df.columns)}")
+            return
+        
```
This function adds the filter columns so no need to check they exist before then.
<!-- Reviewable comment -O_RHUtc15Vt7pGbRyQu:bor2swa -->
<!-- Sent from Reviewable.io -->

RSWilson1 replied:
Yeah this is right it depends in the FILTER_ prefix cols so should change
<!-- Reviewable comment -O_pedyb3O_bGorN3lfj:b-35i043 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Has this been resolved?
<!-- Reviewable comment -O_r5Etc-oMUdekDeOkQ:b-av1r92 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
@coderabbitai[bot], has this been resolved?
<!-- Reviewable comment -O_r5Zv_DJtLypNKS6Z7:b-wpdh0q -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Has this been resolved?
<!-- Reviewable comment -O_r6-kx9-uKFfHjdffe:b-av1r92 -->
<!-- Sent from Reviewable.io -->

RSWilson1 commented on DI-2086/extract_vaf_dp_filter.py:
```diff
@@ -0,0 +1,200 @@
+        )
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] bcftools query failed: {e.stderr}")
+    
```
yes we would prefer a runtime error here as we wouldn't have any data.
<!-- Reviewable comment -O_RAVtp3Z1T8HDhloh7:bj4b4jy -->
<!-- Sent from Reviewable.io -->

RSWilson1 commented on DI-2086/extract_vaf_dp_filter.py:
```diff
@@ -0,0 +1,200 @@
+    df = pd.read_csv(features_file)
+    
+    # Check if required columns exist
+    required_cols = ['REF', 'ALT', 'REF.truth', 'ALT.truth']
+    missing_cols = [col for col in required_cols if col not in df.columns]
+    
+    if missing_cols:
+        print(f"Warning: Missing columns: {missing_cols}")
+        return df
+    
+    # Replace NaN values in REF column with REF.truth
+    df.loc[df['REF'].isna(), 'REF'] = df.loc[df['REF'].isna(), 'REF.truth']
+    
+    # Replace NaN values in ALT column with ALT.truth
+    df.loc[df['ALT'].isna(), 'ALT'] = df.loc[df['ALT'].isna(), 'ALT.truth']
+    
+    # Add the sample name to the records dataframe
+    df['sample'] = sample_name
+    
+    # Return dataframe with dropped duplicates
+    return df.drop_duplicates(subset=['CHROM','POS', 'REF', 'ALT'])
```
Makes sense to add this to list of required cols.
<!-- Reviewable comment -O_RHmpNBFXC5zXR6LYl:b3qt906 -->
<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> @RSWilson1 reviewed 2 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 2 of 3 files reviewed, 25 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-O_R5RuS2-juHpScJS07:b-rpzy96) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I meant it should be  dataframe (pd.DataFrame): The modified dataframe with the new 'FILTER_change'. It would be good to use more specific name than dataframe as well.
___
*[`DI-2086/sompy_plots.py` line 1 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnAi_8CsGlOr6obstA:-OZnAi_8CsGlOr6obstB:b-319s55) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L1)):*
> ```Python
> #!/usr/bin/env python3
> ```

What does this script generate and is there an example?
___
*[`DI-2086/sompy_plots.py` line 8 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_XjV9j0dVnMRCMSo0b:-O_XjV9j0dVnMRCMSo0c:b5ftxtq) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/sompy_plots.py#L8)):*
> ```Python
> @author: arun
> """
> ```

got up to here (reference to self)
___
*[`DI-2086/extract_vaf_dp_filter.py` line 141 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_XhhbNC5y4G0uu9s6-:-O_XhhbNC5y4G0uu9s60:b-a9loko) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/extract_vaf_dp_filter.py#L141)):*
> ```Python
>     
>     # Merge all dataframes
>     merge_df = records_df.merge(query_data_df_renamed, on=merge_cols, how='left')
> ```

because the data should be matching due for records_df vs query_data_df_renamed I'm not sure if join makes a major difference but I think it should be outer here as left will exclude anything just in query_data_df_renamed.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 152 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_RIFcS71pRjk-3sQqZ:-O_RIFcS71pRjk-3sQq_:b76o7ll) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/extract_vaf_dp_filter.py#L152)):*
> ```Python
>         'DP_truth': 0,
>         'VAF_query':0.0,
>         'VAF_truth':0
> ```

should be float i.e. 0.0
___
*[`DI-2086/extract_vaf_dp_filter.py` line 34 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnJ5xE-U7sxIu8cosZ:-OZnJ5xE-U7sxIu8cos_:b-vcusbc) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L34)):*
> ```Python
>     >>> print(len(records))
>     100
>     """
> ```

Remove duplicate entries as features.csv has multiple lines due to same input vcf having multiple duplicate variants due to multiple annotations for more than one transcript.


<!-- Sent from Reviewable.io -->

Commit:
Update read_features_file() docstring

Commit:
extract_vaf_dp_filter.py/extract_variant_data_df() fix raise error

Commit:
extract_vaf_dp_filter.py -> VAF_truth to float

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 2 of 3 files reviewed, 22 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-O_l1b6u-XzF4esKKN0F:b-896fix) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

I meant it should be  dataframe (pd.DataFrame): The modified dataframe with the new 'FILTER_change'. It would be good to use more specific name than dataframe as well.
</blockquote></details>

Done.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 141 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_XhhbNC5y4G0uu9s6-:-O_l5e_l4zrM36XZDq5m:b-kmm693) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/extract_vaf_dp_filter.py#L141)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

because the data should be matching due for records_df vs query_data_df_renamed I'm not sure if join makes a major difference but I think it should be outer here as left will exclude anything just in query_data_df_renamed.
</blockquote></details>

Yes, it wouldn't make any difference id it were run as an outer join. However, when using the outer join, the variant entries are sorted differently. Using a left join creates a dataframe the mirrors the same order as the features.csv file. I also wanted to ensure that the outputs would match with what was previously done in the previous versions of the code. But I am happy to use outer if you prefer.
___
*[`DI-2086/extract_vaf_dp_filter.py` line 152 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_RIFcS71pRjk-3sQqZ:-O_l6WteD34zuAaHIi2g:b-896fix) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/extract_vaf_dp_filter.py#L152)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

should be float i.e. 0.0
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 23 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/sompy_plots.py` line 1 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnAi_8CsGlOr6obstA:-O_l9jACDfJjEmAiK9q9:bxauol3) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L1)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

What does this script generate and is there an example?
</blockquote></details>

Is generates the barplots using the .stats.csv files that are found in a given DNAnexus folderpath.


<!-- Sent from Reviewable.io -->

Commit:
sompy_plots.py -> Replace s1 and s2 statements to truth and query

Commit:
sompy_plots.py -> Deleting unused code

RSWilson1 commented on DI-2086/correlation_plots_pdf_html.py:
```diff
@@ -0,0 +1,273 @@
+                         labels={
+                            'FILTER_change': 'Filter Change'
+                         }
+                         )
```
true there are two colour inputs.
<!-- Reviewable comment -O_ph_ao5hhc0IZPbm-h:b-wqi8i8 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Has this been resolved?
<!-- Reviewable comment -O_r6Akm8TwpBdMuMXD1:bif1t34 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Has this been resolved?
<!-- Reviewable comment -O_r6Hbk0Pio_q8MkA8F:bif1t34 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
@coderabbitai[bot], could you confirm if this has been resolved?
<!-- Reviewable comment -O_r6Omu4rQShjVNJrz3:bnj61f -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
1. df['sample'] has been set as a required column.
2. color_continuous_scale is not there anymore
<!-- Reviewable comment -O_r6uG07EH7ELMbM4Gs:b-u5lmca -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
@coderabbitai[bot] Could you confirm if these points have been applied?
-   df\['sample'\] has been set as a required column.
-   color\_continuous\_scale is not there anymore
<!-- Reviewable comment -O_r7WnnEzoGxHKFZ5Hn:bpn9yqo -->
<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> @RSWilson1 reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 22 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-O_m7kfo58JR04jTFjee:b-40r4rg) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Still looks the same to me?
___
*[`DI-2086/correlation_plots_pdf_html.py` line 39 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_pfKdQ05f52amUogwc:-O_pfKdQ05f52amUogwd:b-d67tfb) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/correlation_plots_pdf_html.py#L39)):*
> ```Python
>                       column.
>     """
>     # Set up all conditions
> ```

Could you add a more detailed comment explaining what is happening here?



_Code quote:_
```Python
onditions
```


<!-- Sent from Reviewable.io -->

Commit:
sompy_plots.py -> remove hardocoded parts and turn script into main()

RSWilson1 committed:
Paired-programming session (extract_vaf_dp.py)
- rm'd whitespace
- set join to 'outer'
- corrected docstring
- Added additional required cols
- Changed error handling of required cols.

Co-authored-by: Arun Karnani Khemlani <Arun-bioinformatics@users.noreply.github.com>

RSWilson1 committed:
Paired programming session
- rm'd whitespace
- Changed input arg to output_path
- Rm'd redundant lines in px code
- Linting changes
- now outputs plot dataframe as well.
Co-authored with Arun

RSWilson1 committed:
Paired-programming session
- Fixed required columns
Co-Authored with Arun

RSWilson1 committed:
Paried programming
- rm'd whitespace
Co-authored with Arun

RSWilson1 commented on DI-2086/extract_vaf_dp_filter.py:
```diff
@@ -0,0 +1,201 @@
+    if not vcf_path:
+        raise FileNotFoundError(f"VCF file not identified in {vcf_path}")
+
+    # Run the bcftools command
+    try:
+        result = subprocess.run(
+            [
+                "bcftools",
+                "query",
+                "-f",
+                "'%CHROM\t%POS\t%REF\t%ALT\t[%DP]\t[%VF]\t%FILTER\n'",
+                vcf_path
+            ],
```
Not needed ignore
<!-- Reviewable comment -O_qPz2q1NdDVLyAzwzt:bgto2h7 -->
<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (commented):
> @RSWilson1 reviewed 2 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 13 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/extract_vaf_dp_filter.py` line 34 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZnJ5xE-U7sxIu8cosZ:-O_qObgZBfxLSQpVgvNe:b5vbbjy) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/extract_vaf_dp_filter.py#L34)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Remove duplicate entries as features.csv has multiple lines due to same input vcf having multiple duplicate variants due to multiple annotations for more than one transcript.
</blockquote></details>

added in paired programming session
___
*[`DI-2086/sompy_plots.py` line 14 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihrjsCTAYnWJPOOPt:-O_qbkbP5B-ktM7PvNwd:bujqfi5) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L14)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

hard-coded
</blockquote></details>

resolved
___
*[`DI-2086/sompy_plots.py` line 19 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihu5S-zrYAKymzwqm:-O_qbnWr7IiIZdHWkYVF:bujqfi5) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L19)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

hard-coded
</blockquote></details>

resolved
___
*[`DI-2086/sompy_plots.py` line 27 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZihuru0_BM-PXyw_Bg:-O_qbpEl3DipeHRVH4-p:bujqfi5) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/sompy_plots.py#L27)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

hard-coded
</blockquote></details>

resolved


<!-- Sent from Reviewable.io -->

Arun-bioinformatics commented on DI-2086/correlation_plots_pdf_html.py:
```diff
@@ -0,0 +1,210 @@
+import argparse
+import glob
+import numpy as np
+import pandas as pd
+from pathlib import Path
+import plotly.express as px
+import plotly.graph_objects as go
+
+def add_filter_column_change(dataframe):
+    """
+    Add a filter change column to the dataframe based on the conditions specified.
+    The new column 'FILTER_change' will indicate:
+    - 'No change PASS' if both FILTER_s1 and FILTER_s2 contain 'PASS'
+    - 'No change EXCLUDE' if both FILTER_s1 and FILTER_s2 contain 'EXCLUDE'
+    - 'Change from PASS to EXCLUDE' if FILTER_s1 contains 'PASS' and FILTER_s2 contains 'EXCLUDE'
+    - 'Change from EXCLUDE to PASS' if FILTER_s1 contains 'EXCLUDE' and FILTER_s2 contains 'PASS'
+    - 'Variant removed PASS' if FILTER_s1 contains 'PASS' and FILTER_s2 is '.'
+    - 'Variant removed EXCLUDE' if FILTER_s1 contains 'EXCLUDE' and FILTER_s2 is '.'
+    - 'Variant added PASS' if FILTER_s1 is '.' and FILTER_s2 contains 'PASS'
+    - 'Variant added EXCLUDE' if FILTER_s1 is '.' and FILTER_s2 contains 'EXCLUDE'
+
+    Args:
+        dataframe (pd.DataFrame): The input dataframe containing 'FILTER_s1' and 'FILTER_s2' columns.
+
+    Returns:
+        pd.DataFrame: The modified dataframe with the new 'FILTER_change' column.
+    """
+    conditions = [
+        (dataframe['FILTER_s1'].str.contains('PASS')) & (dataframe['FILTER_s2'].str.contains('PASS')),
+        (dataframe['FILTER_s1'].str.contains('EXCLUDE')) & (dataframe['FILTER_s2'].str.contains('EXCLUDE')),
+        (dataframe['FILTER_s1'].str.contains('PASS')) & (dataframe['FILTER_s2'].str.contains('EXCLUDE')),
+        (dataframe['FILTER_s1'].str.contains('EXCLUDE')) & (dataframe['FILTER_s2'].str.contains('PASS')),
+        (dataframe['FILTER_s1'].str.contains('PASS')) & (dataframe['FILTER_s2'] == '.'),
+        (dataframe['FILTER_s1'].str.contains('EXCLUDE')) & (dataframe['FILTER_s2'] == '.'),
+        (dataframe['FILTER_s1'] == '.') & (dataframe['FILTER_s2'].str.contains('PASS')),
+        (dataframe['FILTER_s1'] == '.') & (dataframe['FILTER_s2'].str.contains('EXCLUDE')),
+    ]
```
What if I would like it to raise on NaNs?
<!-- Reviewable comment -O_r4jRM0MCnMtQElUxz:b-k59qvv -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 13 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 39 at r3](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_pfKdQ05f52amUogwc:-O_r4_Eh-hG-7y-XqH_2:b-896fix) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/d561f95ba84cf1ad7b633180d5de18ca380fbe23/DI-2086/correlation_plots_pdf_html.py#L39)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Could you add a more detailed comment explaining what is happening here?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 12 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 12 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 12 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 12 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 12 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 12 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 11 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 11 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (commented):
> @RSWilson1 reviewed 2 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 2 of 3 files reviewed, 8 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)



<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 2 of 3 files reviewed, 9 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/sompy_plots.py` line 138 at r7](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_uwZOFEQcnQTFWGspR:-O_uwZOFEQcnQTFWGspS:b-8xb4k2) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/7fc129d692828f64f756c5a012df911d86fe796c/DI-2086/sompy_plots.py#L138)):*
> ```Python
> 
>     # Save the Dataframe in a .csv file
>     stats_df.to_csv('sompy_plots_table.csv')
> ```

Make this configurable


<!-- Sent from Reviewable.io -->

Commit:
Minor changes in correlations_plot_pdf_html.py:
-  Changed the input var for add_filter_column_change()
- Implement a raise error when saving pdfs if kaleido is not installed

Commit:
Major fixes in sompy_plots.py
-  Added "output_path" as input. Should be used to specify where the output files will be stored (plots and .csv table)
- generate_barplot_from_DNAnexus_path() has been renamed to generate_stats_df_from_DNAnexus_path()
- Logic generating the plots has been shifted to generate_plots_from_stats_df()
- Each generated plot is now stored in a .png file, output path is now configurable

Arun-bioinformatics commented on DI-2086/sompy_plots.py:
```diff
@@ -0,0 +1,105 @@
+    name_mode="glob",
+    project=project,
+    folder="/query_filtered_VCFs_removed_sompy_results"
+))
```
Hard-coded lines removed, so resolved
<!-- Reviewable comment -O_w7M3v9w89xdSuEASq:b-2aeonl -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics commented on DI-2086/sompy_plots.py:
```diff
@@ -0,0 +1,105 @@
+    unique_to_S2.append(df[df['type'] == 'records']['fp'].iloc[0])
+    # Count the munber of variants found only in S1 (fn)
+    unique_to_S1.append(df[df['type'] == 'records']['fn'].iloc[0])
+
```
I would like it to raise if the row is absent
<!-- Reviewable comment -O_w7UXi4diqMcllg3Lj:b-t7muxv -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics commented on DI-2086/sompy_plots.py:
```diff
@@ -0,0 +1,105 @@
+stats_df['shared'] = shared_variants
+stats_df['S1 only'] = unique_to_S1
+stats_df['S2 only'] = unique_to_S2
+
```
issue resolved? could could you confirm @coderabbitai[bot] ?
<!-- Reviewable comment -O_w7YxX4GXSll5602DM:b-rzxc8r -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Script works fine where a set is given as a fields argument. If there is no NaN handling, would it then raise an error? if that is so I would like to keep this feature.
<!-- Reviewable comment -O_w8aLO2qm4Z0U8oUgX:b-ttnfl6 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 8 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-O_w6Al5-9p68MOI4a-Z:by4ikcx) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Still looks the same to me?
</blockquote></details>

I wasn't really sure of what was the required change. However, I have now made the arg more specific to indicate what kind of dataframe is required.


<!-- Sent from Reviewable.io -->

Arun-bioinformatics commented on DI-2086/sompy_plots.py:
```diff
@@ -0,0 +1,105 @@
+dxfile = dxpy.DXFile('file-J24FXZQ48fFz1gb8J1Jxg1Vq').read()
+type(dxfile)
+df = pd.read_csv(StringIO(dxfile))
+
```
is this resolved @coderabbitai[bot] 
<!-- Reviewable comment -O_w7pgX95ra5BkljnqB:bwrui87 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
This has been already addressed
<!-- Reviewable comment -O_w9OMf7ix7kRUNF1Fh:b-zen5ye -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 8 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/sompy_plots.py` line 138 at r7](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-O_uwZOFEQcnQTFWGspR:-O_w8sTf0kUiK7kjnxbk:bhglw15) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/7fc129d692828f64f756c5a012df911d86fe796c/DI-2086/sompy_plots.py#L138)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Make this configurable
</blockquote></details>

It is now configurable


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 4 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Commit:
Correlation_plots_pdf_html -> pearson correlation line renamed to y=x reference line

RSWilson1 commented on DI-2086/correlation_plots_pdf_html.py:
```diff
@@ -0,0 +1,285 @@
+    """
+    # Set up all conditions
+    conditions = [
+        # No change PASS
+        (merged_var_df["FILTER_truth"].str.contains("PASS"))
+        & (merged_var_df["FILTER_query"].str.contains("PASS")),
+        # No change EXCLUDE
+        (merged_var_df["FILTER_truth"].str.contains("EXCLUDE"))
+        & (merged_var_df["FILTER_query"].str.contains("EXCLUDE")),
+        # Change from PASS to EXCLUDE
+        (merged_var_df["FILTER_truth"].str.contains("PASS"))
+        & (merged_var_df["FILTER_query"].str.contains("EXCLUDE")),
+        # Change from EXCLUDE to PASS
+        (merged_var_df["FILTER_truth"].str.contains("EXCLUDE"))
+        & (merged_var_df["FILTER_query"].str.contains("PASS")),
+        # Variant removed PASS
+        (merged_var_df["FILTER_truth"].str.contains("PASS"))
+        & (merged_var_df["FILTER_query"] == "."),
+        # Variant removed EXCLUDE
+        (merged_var_df["FILTER_truth"].str.contains("EXCLUDE"))
+        & (merged_var_df["FILTER_query"] == "."),
+        # Variant added PASS
+        (merged_var_df["FILTER_truth"] == ".")
+        & (merged_var_df["FILTER_query"].str.contains("PASS")),
+        # Variant added EXCLUDE
+        (merged_var_df["FILTER_truth"] == ".")
+        & (merged_var_df["FILTER_query"].str.contains("EXCLUDE")),
+        # No change, rescued variant
+        (
+            merged_var_df["FILTER_truth"].str.contains("rescued")
+            | merged_var_df["FILTER_query"].str.contains("rescued")
+        )
+        & (merged_var_df["FILTER_truth"] == merged_var_df["FILTER_query"]),
+        # Changed rescued variant
+        (
+            merged_var_df["FILTER_truth"].str.contains("rescued")
+            | merged_var_df["FILTER_query"].str.contains("rescued")
+        )
+        & (merged_var_df["FILTER_truth"] != merged_var_df["FILTER_query"]),
+    ]
```
for rescued variants do they include PASS or EXCLUDE?
<!-- Reviewable comment -O_wKU8WA3EDw7HIQGax:bwzmz2t -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
I had a look at the data, there hasn't been cases that are similar to "rescued,PASS" but there are some cases that are similar to "rescued, EXCLUDE". However, for any variants that have a "rescued" string regardless of whether they also contain "PASS" or "EXCLUDE". I wanted these to be annotated differently to the ones that are not "rescued". Hence why I kept the order of conditions as it is. @coderabbitai[bot], could you confirm that this order would be okay?
<!-- Reviewable comment -O_wV0tM-hc_khL5xdBg:b-uwm5p4 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
I see, what if for the variants that contain EXCLUDE on both "FILTER_truth" and "FILTER_query", I want these to be annotated as "No change EXCLUDE". Would the order of the conditions be okay?
<!-- Reviewable comment -O_wY3fO-ds5fuKqMFG3:b9g37ta -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
@coderabbitai[bot] 
<!-- Reviewable comment -O_wYe3EAUYF0CrsR_G3:b-aqk4sv -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
We have decided to keep the order as finding EXCLUDE in both truth and query takes priority
<!-- Reviewable comment -Oa-_Yi6ASsN03Y68uWe:b-5bcbwp -->
<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> @RSWilson1 reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-O_wJYwRDyFk_WvhCD4c:b8kqnfs) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

I wasn't really sure of what was the required change. However, I have now made the arg more specific to indicate what kind of dataframe is required.
</blockquote></details>

in the returns can you add the variable name you are returning please?


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 26 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-OZibXLf994j-etY7zBh:-O_wSapb-mKHevnYZl4Y:b-896fix) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/6c1c9f79388e296192a4698f4f04afaf091035ee/DI-2086/correlation_plots_pdf_html.py#L26)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

in the returns can you add the variable name you are returning please?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Commit:
correlation_plots_pdf_html.py -> included variable name in Docstring Return

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @mattgarner and @RSWilson1)



<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @Arun-bioinformatics and @mattgarner)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 41 at r10](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-Oa-982EEQBd-jw0jizE:-Oa-982EEQBd-jw0jizF:bu7u8re) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/0e0fdd5a31a4bd898006819e119d77a6d9b5641f/DI-2086/correlation_plots_pdf_html.py#L41)):*
> ```Python
>                                            column.
>     """
>     # Set up all conditions
> ```

Could you explain the logic here and how exclude overrides the rescued part


<!-- Sent from Reviewable.io -->

Commit:
Required=True on all scripts and examples for each condition
- Minor fixes in correlation_plots_pdf_html.py and sompy_plots.py

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @mattgarner and @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 41 at r10](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3#-Oa-982EEQBd-jw0jizE:-Oa-_jeY00e6TNKxGQ8w:bd713jl) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/0e0fdd5a31a4bd898006819e119d77a6d9b5641f/DI-2086/correlation_plots_pdf_html.py#L41)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

Could you explain the logic here and how exclude overrides the rescued part
</blockquote></details>

Examples provided for each condition. Notes section provided to describe how the function works on edge cases.


<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (approved):
> @RSWilson1 reviewed 3 of 3 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/3)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @mattgarner)



<!-- Sent from Reviewable.io -->
